File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        mej_elem = None
        me_cnt = 0

        for num in nums:
            if me_cnt == 0:
                mej_elem = num
                me_cnt = 1  
            elif num == mej_elem:
                me_cnt += 1
            else:
                me_cnt -= 1

        return mej_elem

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let majorityElement = nums[0];

    let voteCount = 1;

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] === majorityElement) {
            voteCount += 1;
        } else {
            voteCount -= 1
        }

        if (voteCount === 0) {
            majorityElement = nums[i];
            voteCount = 1;
        }
    }

    return majorityElement;


};